plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/editor/context/project/ProjectContextProvider.kt

```diff
@@ -0,0 +1,355 @@
+    )
+
+    private fun initAndIndex() {
+        scope.launch {
+            while (retryCount.get() < 5) {
+                try {
+                    logger.info { "project context: about to init key" }
+                    val isInitSuccess = initEncryption()
+                    if (isInitSuccess) {
+                        logger.info { "project context index starting" }
+                        delay(300)
+                        val isIndexSuccess = index()
+                        if (isIndexSuccess) isIndexComplete.set(true)
+                        return@launch
+                    }
+                } catch (e: Exception) {
+                    if (e.stackTraceToString().contains("Connection refused")) {
+                        retryCount.incrementAndGet()
+                        delay(10000)
+                    } else {
+                        return@launch
+                    }
+                }
+            }
+        }
+    }
+
+    private fun initEncryption(): Boolean {
+        logger.info { "project context: init key for ${project.guessProjectDir()} on port ${encoderServer.port}" }
+        val url = URL("http://localhost:${encoderServer.port}/initialize")
+        val payload = encoderServer.getEncryptionRequest()
+        val connection = url.openConnection() as HttpURLConnection
+        setConnectionProperties(connection)
+        setConnectionRequest(connection, payload)
+        logger.info { "project context initialize response code: ${connection.responseCode} for ${project.name}" }
+        return connection.responseCode == 200
+    }
+
+    fun index(): Boolean {
+        logger.info { "project context: indexing ${project.name} on port ${encoderServer.port}" }
+        val indexStartTime = System.currentTimeMillis()
+        val url = URL("http://localhost:${encoderServer.port}/indexFiles")
+        val filesResult = collectFiles()
+        val projectRoot = project.guessProjectDir()?.path ?: return false
+        val payload = IndexRequestPayload(filesResult.files, projectRoot, true)
+        val payloadJson = mapper.writeValueAsString(payload)
+        val encrypted = encoderServer.encrypt(payloadJson)
+
+        val connection = url.openConnection() as HttpURLConnection
+        setConnectionProperties(connection)
+        setConnectionRequest(connection, encrypted)
+        logger.info { "project context index response code: ${connection.responseCode} for ${project.name}" }
+        val duration = (System.currentTimeMillis() - indexStartTime).toDouble()
+        val startUrl = getStartUrl(project)
+        if (connection.responseCode == 200) {
+            val usage = getUsage()
+            TelemetryHelper.recordIndexWorkspace(duration, filesResult.files.size, filesResult.fileSize, true, usage?.memoryUsage, usage?.cpuUsage, startUrl)
+            logger.info { "project context index finished for ${project.name}, list of files indexed: ${filesResult.files.joinToString(",")}" }
```

Comment:
```suggestion
logger.info { 
    "project context index finished for ${project.name}, number of files indexed: ${filesResult.files.size}" 
}
```

Reply:
just realized now 
![image](https://github.com/aws/aws-toolkit-jetbrains/assets/96078566/a7881279-fcb8-411b-981f-49c38ba70e9c)

Reply:
this was a requirement to log all index files, maybe the formatting can be better?

Reply:
actually I will change this to debug

Reply:
> this was a requirement to log all index files, maybe the formatting can be better?
isee,  use simple file name maybe?

Reply:
and yes debug level make sense to me